app/services/VirtualMachineService.ts

```diff
@@ -0,0 +1,584 @@
+import { PrismaClient, VM } from ".prisma/client";
+import { ExerciseService } from "./ExerciseService";
+import axios from "axios";
+import { config } from "../../config";
+import InsufficientStorageException from "../exceptions/InsufficientStorageException";
+import VirtualMachineConflictException from "../exceptions/VirtualMachineConflictException";
+
+const prisma = new PrismaClient();
+
+const minVMID = 400;
+const maxVMID = 499;
+
+export const VirtualMachineService = {
+  /**
+   * Creates the VM in proxmox
+   *
+   * @param {string} user current user
+   * @param {string} exerciseId vmid of exercise
+   * @param {string} nodeName node name to manual assign load
+   * @return error from function
+   */
+  async createVM(
```

Comment:
Make sure to validate the node sent by the user.

I want to maintain an array of "whitelisted" nodes on the backend. Those are the ONLY nodes the client is aware of. 

Which means on the frontend, when you select a node to launch the VM on, it should source those nodes from this list of whitelisted nodes.

Reply:
I also want aliases.